gym/f110_gym/envs/f110_env.py

```diff
@@ -311,6 +322,48 @@ def reset(self, poses):
         obs, reward, done, info = self.step(action)
         return obs, reward, done, info
 
+    def add_obstacles(self, n_obstacles=4, obstacle_size=[0.5, 0.5]):
+        """
+        Adds a set number of obstacles to the envioronment. 
+        Updates the renderer and the map kept by the laser scaner for each vehicle in the simulator
+
+        Args:
+            n_obstacles (int): number of obstacles to add
+            obstacle_size (list(2)): rectangular size of obstacles
+            
+        Returns:
+            None
+        """
+        if self.renderer is None:
+            self.render()
+        map_img = np.copy(self.empty_map_img)
+
+        obs_size_m = np.array(obstacle_size)
+        obs_size_px = obs_size_m / self.map_resolution
+
+        obs_locations = []
+        while len(obs_locations) < n_obstacles:
+            rand_x = int(np.random.random() * (self.map_width - obs_size_px[0]))
+            rand_y = int(np.random.random() * (self.map_height - obs_size_px[1]))
+
+            if self.original_dt[rand_y, rand_x] > 0.05:
```

Comment:
With this method of checking how close to a wall, you don't have a guarantee whether the obstacle is inside or outside the track. Also, this might result in an infinite loop.

Reply:
This is a valid issue. I was working on maps like Torino where the only driveable area is on the track, but I can see how this could be a problem.
Adding a method that places obstacles according to the centerline is probably the best way to combat this.